src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,140 @@
+        return finalString;
+    }
+
+    private static String processorFind(String command, TaskList taskList) {
+        int counter = 1;
+        try {
+            String[] stringarr = command.split(" ", 2);
+            String key = stringarr[1];
+            if (key.length() == 0) {
+                return Ui.showError("Please provide a key");
+            }
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.append("Here are the matching tasks in your list: \n");
+            for (int i = 0; i < taskList.getListSize(); i++) {
+                if (taskList.getTask(i).getTask().contains(key)) {
+                    String findResponse = counter + "." + taskList.getTask(i).toString() + "\n";
+                    stringBuilder.append(findResponse);
+                    counter++;
+                }
+            }
+            stringBuilder.append(Ui.showLine());
+            return stringBuilder.toString();
+        } catch (Exception e) {
+            return Ui.showError("Please give a key");
+        }
+    }
+
+    private static String processorList(TaskList taskList) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Here are the tasks in your list: \n");
+        for (int i = 0; i < taskList.getListSize(); i++) {
+            int index = i + 1;
+            String listResponse = index + "." + taskList.getTask(i).toString() + "\n";
+            stringBuilder.append(listResponse);
+        }
+        stringBuilder.append(Ui.showLine());
+        return stringBuilder.toString();
+    }
+
+    private static String processorAdd(String cmd, TaskList taskList) throws DukeException {
+        String[] stringarr = cmd.split(" ", 2);
+        if (stringarr[0].equals("todo")) {
+            if (stringarr.length <= 1) {
+                String message = "The description of a Todo cannot be empty";
+                throw new DukeException(message);
+            } else {
+                Todo todo = Todo.createTodo(stringarr[1]);
+                taskList.addTask(todo);
+                return todo.toString();
+            }
+        } else if (stringarr[0].equals("deadline")) {
+            if (stringarr.length <= 1) {
+                String message = "The description of a Deadline cannot be empty";
+                throw new DukeException(message);
+            } else {
+                try {
+                    String[] secondarr = stringarr[1].split("/by", 2);
+                    LocalDate date = LocalDate.parse(secondarr[1].trim());
+                    Deadline deadline = Deadline.createDeadline(secondarr[0], date);
+                    taskList.addTask(deadline);
+                    return deadline.toString();
+                } catch (DateTimeParseException e) {
+                    String message = "That does not look like a proper Date. Please input YYYY-MM-DD";
+                    throw new DukeException(message);
+                }
+            }
```

Comment:
Perhaps you can restructure this code block to make the happy path more prominent?